client/my-sites/site-settings/controller.js

```diff
@@ -78,27 +77,18 @@ module.exports = {
 		}
 
 		// if user went directly to jetpack settings page, redirect
-		if ( site.jetpack && ! config.isEnabled( 'manage/jetpack' ) ) {
+		if ( site && site.jetpack && ! config.isEnabled( 'manage/jetpack' ) ) {
 			window.location.href = '//wordpress.com/manage/' + site.ID;
 			return;
 		}
 
-		if ( ! site.latestSettings || new Date().getTime() - site.latestSettings > ( fiveMinutes ) ) {
-			if ( sites.initialized ) {
-				site.fetchSettings();
```

Comment:
One thing that I almost missed here: if we're removing the code that fetches the site settings, let's make sure we're querying settings in the affected components (that might require adding a `<QuerySiteSettings />` here and there).